chain/neutrino.go

```diff
@@ -815,3 +815,22 @@ out:
 	close(s.dequeueNotification)
 	s.wg.Done()
 }
+
+// MapRPCErr takes an error returned from calling RPC methods from various
+// chain backend and map it to an defined error here. It uses the `BtcdErrMap`,
+// whose keys are btcd error strings and values are errors made from bitcoind
+// error strings.
+//
+// NOTE: we assume neutrino shares the same error strings as btcd.
+func (s *NeutrinoClient) MapRPCErr(rpcErr error) error {
+	// Iterate the map and find the matching error.
+	for btcdErr, matchedErr := range BtcdErrMap {
+		// Match it against btcd's error.
+		if matchErrStr(rpcErr, btcdErr) {
+			return matchedErr
+		}
+	}
+
+	// If not matched, return the original error wrapped.
+	return fmt.Errorf("%w: %v", ErrUndefined, rpcErr)
```

Comment:
Q: Shouldn't we wrap it the other way around or even both ?

Reply:
`ErrUndefined` is the defined type tho, so in `lnd` when we see `chain.ErrUndefined` we know it's an undefined error. On the other hand `rpcErr` won't have a type in this case, also we cannot use more than one `%w` in `fmt.Errorf` unfortunately.